### PR TITLE
Rewrite for better support of UUID

### PR DESCRIPTION
**Pull request by LukeHandle**
Will be merging this in to resolve issues facing live site https://minotar.net/ (old skins returned).

We update imgd to use new methods as well.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 language: go
 
 go:
-  - 1.1
```
Comment (connor4312) [inline .travis.yml]: What a throwback 😄

**Comment by connor4312** [comment]
I'll give this a full review this evening, don't feel you need to block on me though

**Comment by LukeHandle** [comment]
I've raised https://github.com/minotar/imgd/pull/182 to make use of some of the functions added. 

Please ignore `FetchTextures()`. It's a mess and I do not make use of it in the above PR (note the commit data here :D )

Thanks @connor4312  !

**Comment by LukeHandle** [comment]
Merging in. Will need some touching up in the future.

**Pull request merged**